### ISSUE TITLE
[cli] Install packages as well when using `generate` command

### ISSUE THREAD

**Issue by sooraj1002** [SamagraX-Stencil/stencil-cli#8]
## Description

Presently, the cli while setting up a service using the `generate` command instead of setting it up during the bootstrapping phase does not install the external/extra dependencies if any. This should not be the case, those should be added alongside the service setup.

## Potential fix
1. Detect the `lock` file in the package root, if the package root contains only `yarn.lock` or `pnpm.lock` use `yarn` or `pnpm` respectively, incase there is no lock file of multiple lock files default to `npm`.
2. Read, parse and update the `package.json` to add the dependency with the particular version that is required and print a message for the user to install the new dependencies using a package manager of their choice.

## Objective
The goal of this issue is to update the package.json when user runs the `generate` command and prompt user on which package manager he needs to install the packages or use the registry generated here https://github.com/SamagraX-Stencil/stencil-cli/pull/20 to know which package manager the user has chosen while bootstraping the project.

## Goals

- [ ] Identify the package manager (package.json/yarn.lock/pnpm.lock/bun.lockb)
- [ ] Update the specified package.json with necessary packages name
- [ ]  Install the packages
- [ ] Write test for this function

Before `generate` command                                       |  After `generate` command
:--------------------------------------------------------:|:-----------------------------------------------:

<p float="left">
  <img src="https://github.com/SamagraX-Stencil/stencil-cli/assets/91362589/7b0d5d8f-3b3f-4375-b2f7-1ea87f91903c" width="350" /> 
  <img src="https://github.com/SamagraX-Stencil/stencil-cli/assets/91362589/6ddec09b-23d9-497a-aad4-88f7449428b6" width="350" />
</p>

### Product Name
Stencil-Cli

## Organisation Name
SamagraX

## Domain
Command Line Interface

## Tech Skills Needed
TypeScript, Cli

**Comment by Savio629** [comment]
@sooraj1002 
So instead of installing it directly, we will have to store it in dependencies and then prompt user for package manager of his choice to install the dependencies?
Can't we just install it when the specific module is being generated?

**Comment by sooraj1002** [comment]
- It would be better to update the package.json manually, so that in case of any failures, user can manually just run an install
- When the user already has a `yarn.lock` or `pnpm.lock` file, we can directly assume thats the package manager we have to use
- if we dont have those lock files, then we can ask the user. and in case of bootstrapping a new project, just use the package manager selected

@techsavvyash can you also confirm this once?

**Comment by techsavvyash** [comment]
Yes, this is the correct approach, incase a direct install fails, it will not update the `package.json` making it very difficult for the end user to figure out what all dependencies are required giving a very bad UX, we must first update the `package.json` manually, which should be easy work, then we should check for the existence of lock files, incase of conflict, just prompt the user for the package manager and use that.
Additionally the registry feature that @Savio629 added can be used to store the package manager of user's choice as well removing the need to check for package manager over and over again.

**Comment by Savio629** [comment]
### Description
Presently, the cli while setting up a service using the generate command instead of setting it up during the bootstrapping phase does not install the external/extra dependencies if any. This should not be the case, those should be added alongside the service setup.


### Objective
The goal of this issue is to update the package.json when user runs the `generate` command and prompt user on which package manager he needs to install the packages or use the registry generated here https://github.com/SamagraX-Stencil/stencil-cli/pull/20 to know which package manager the user has chosen while bootstraping the project.

### Goals

- [ ] Identify the package manager (package.json/yarn.lock/pnpm.lock/bun.lockb)
- [ ] Update the specified package.json with necessary packages name
- [ ]  Install the packages
- [ ] Write test for this function


Before `generate` command                                       |  After `generate` command
:--------------------------------------------------------:|:-----------------------------------------------:



<p float="left">
  <img src="https://github.com/SamagraX-Stencil/stencil-cli/assets/91362589/7b0d5d8f-3b3f-4375-b2f7-1ea87f91903c" width="350" /> 
  <img src="https://github.com/SamagraX-Stencil/stencil-cli/assets/91362589/6ddec09b-23d9-497a-aad4-88f7449428b6" width="350" />
</p>

#### Complexity : Medium


#### Product Name
Stencil-Cli

### Organisation Name
SamagraX

### Domain
Command Line Interface

### Tech Skills Needed
TypeScript, Cli

**Comment by shivankurchavan** [comment]
please assign me : )
